### PR TITLE
First Sinopia Profile Editor integration testing with jest-puppeteer 

### LINKED ISSUE #23: test for rendering UI with angular, etc.
on laptops and circleci

- test angular rendering of profile editor

-------
- test that loading profile is possible (#24) --> possibly postponed until we are loading profiles our way (or possibly initial test that shows a profile is loaded somehow)
- test that saving profile is possible -- ditto (as part of #4 ?)

### LINKED ISSUE #33: use jest-puppeteer for integration tests
Exemplars:
- https://github.com/ProjectMirador/research-and-demos/tree/master/minimal_redux_poc (Uses webpack, jest, puppeteer, react, enzyme)
- https://github.com/UniversalViewer/universalviewer (Uses grunt + jest + puppeteer)

### PR DESCRIPTION
**Pull request by jermnelson**
This PR adds a simple Jest integration test using puppeteer. Running `npm test` from the source directory will launches a simple web browser and test for "Profiles on Server" being render by the Angular client application. Tagging this WIP because failing on circle-ci build.

 
Fixes #23 
Fixes #33

### CODE REVIEW

File: source/package.json [hunk 1]
```diff
@@ -1,8 +1,7 @@
 {
   "name": "sinopia_profile_editor",
-  "description": "Editor for BIBFRAME *profiles* forked from http://github.com/lcnetdev/profile-edit",
-  "version": "0.0.1",
```
Comment (ndushay) [inline source/package.json]: this is the right description and version -- the one in master.

**Comment by ndushay** [comment]
One fix (aside from rebasing on master) to get it working in circleci is to make these two changes (see LD4P/sinopia_editor#31)

- Add `.circleci/setup_puppeteer.sh` to load a bunch of libraries missing from circleci docker image
- add run step for that script in `.circleci/config.yml` before npm install step

**Comment by jermnelson** [comment]
Thanks @ndushay!  doing your recommendations now; helpful to go through your Sinopia Editor PR 31 (https://github.com/LD4P/sinopia_editor/pull/31/)  on a similar issue.

**Comment by ndushay** [comment]
Looks like you need to do this now:  https://github.com/LD4P/sinopia_editor/blob/master/package.json#L11-L13

File: source/jest-puppeteer.config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  launch: {
+    headless: process.env.CI === 'true'
+  },
+  server: {
+    // TODO: insure http-server is in path
+    command: 'node server.js',
+    port: 8000,
+    launchTimeout: 4000
```
Comment (jgreben) [inline source/jest-puppeteer.config.js]: I'm not sure if we really need this server config.

File: source/package.json [hunk 3]
```diff
@@ -28,18 +28,29 @@
     "grunt-ng-annotate": "^3.0.0",
     "grunt-ngdocs": "^0.2.11",
     "grunt-plato": "^1.4.0",
-    "jest": "^23.6.0"
+    "http-server": "^0.11.1",
```
Comment (jgreben) [inline source/package.json]: Do we still need `http-server` if we are using express?

File: source/jest-puppeteer.config.js (hunk 2 above)
Comment (jgreben) [reply source/jest-puppeteer.config.js]: The default is 5000, so IMO we can remove this.

Comment (jermnelson) [reply source/jest-puppeteer.config.js]: Will remove.

File: source/package.json (hunk 3 above)
Comment (jermnelson) [reply source/package.json]: Will remove as well

File: source/__tests__/locApp.test.js [hunk 4]
```diff
@@ -1,5 +1,11 @@
-const locApp = require("../assets/js/ng-app.js")
+describe('Sinopia Profile Editor Homepage', () => {
 
-test('locApp can be loaded', () => {
+  beforeAll(async () => {
+    await page.goto('http://127.0.0.1:8000')
+  }, 100000)
```
Comment (ndushay) [inline source/__tests__/locApp.test.js]: can we get rid of these timeouts now?  (reviewing by commit, so ignore if addressed in subsequent commit)

File: source/package.json (hunk 1 above)
Comment (ndushay) [reply source/package.json]: Not sure why this fuller, more accurate description is getting stomped on.  Perhaps it needs some manual love (a commit that is squashed into this one?)

File: source/__tests__/locApp.test.js [hunk 5]
```diff
@@ -2,10 +2,10 @@ describe('Sinopia Profile Editor Homepage', () => {
 
   beforeAll(async () => {
     await page.goto('http://127.0.0.1:8000')
-  }, 100000)
+  }, 10000)
 
   it('displays "Profiles on Server" text on page', async () => {
     await expect(page).toMatch('Profiles on Server')
-  }, 100000)
+  }, 10000)
```
Comment (ndushay) [inline source/__tests__/locApp.test.js]: yeah - same question about these timeouts - do we still need?

File: source/__tests__/locApp.test.js (hunk 4 above)
Comment (jgreben) [reply source/__tests__/locApp.test.js]: The default is 500, so I'm ok with removing 10000. If we run into problems with timeouts we can change it back where needed.

File: source/__tests__/locApp.test.js (hunk 5 above)
Comment (jgreben) [reply source/__tests__/locApp.test.js]: The default timeout is 500, so I agree to remove these for now. If we run into problems we can add them back where needed. Also I agree with squashing the commits.

**Comment by ndushay** [comment]
Just a commit squashing away from a merge!

**Comment by jermnelson** [comment]
Done!

**Pull request merged**